composer.json

```diff
@@ -44,7 +44,7 @@
         "http-interop/http-factory-tests": "^0.5.0",
         "laminas/laminas-coding-standard": "~1.0.0",
         "php-http/psr7-integration-tests": "^1.0",
-        "phpunit/phpunit": "^7.5.18"
+        "phpunit/phpunit": "^7.5.18 || ^9.3"
```

Comment:
Should probably first upgrade the test suite

Reply:
Currently diactoros is PHP 7.1+ while phpunit 9.3 is PHP 7.3+, so I was trying to see if both phpunit versions can be supported without bumping up diactoros' PHP version constraint :slightly_smiling_face:.

Reply:
Yeah, we can bump to `^7.3` in the next minor

Reply:
Currently CakePHP 4 supports PHP 7.2+ and has diactoros as dependency. So unless really necessary we would prefer if diactoros does not goto 7.3+ just due to phpunit :)

Supporting phpunit 8.5 and 9.3 together with php 7.2+ isn't too difficult, we have done it https://github.com/cakephp/cakephp/pull/14885.

Reply:
Cake can support an older version of diactoros, and then you'll be fine 👍

Reply:
Only if the older version supports PHP 8 too.

Reply:
Why would it need to? If you pin to `^2.3.1`, you get `7.1.*` support, as well as PHP 8 support (which will happen with `2.4.0`, according to this patch)